### PR TITLE
[stable/nats] Adapt chart to NATS version 2

### PR DESCRIPTION
**Pull request by tompizmor**
#### What this PR does / why we need it:
In NATS v2 the main server binary has been renamed from `gnastd` to `nats-server`. This PR makes the necessary changes.

#### Special notes for your reviewer:
Don't merge yet.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [X] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [X] Chart Version bumped
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

**Comment by javsalgar** [comment]
I see that the changes to make it work with the new version or nats are mininal. Could we parametrize the config file name and the command so it is compatible with the previous version of nats?

**Comment by tompizmor** [comment]
@javsalgar I implemented the requested change. Please take a look when you have a chance

File: stable/nats/templates/deployment.yaml [hunk 1]
```diff
@@ -128,8 +128,8 @@ spec:
 {{ toYaml .Values.resources | indent 10 }}
         volumeMounts:
           - name: config
-            mountPath: /opt/bitnami/nats/gnatsd.conf
-            subPath: gnatsd.conf
+            mountPath: /opt/bitnami/nats/nats-server.conf
```
Comment (javsalgar) [inline stable/nats/templates/deployment.yaml]: We need to change this as well, right?

**Review by javsalgar — CHANGES_REQUESTED** [review]

File: stable/nats/templates/deployment.yaml (hunk 1 above)
Comment (tompizmor) [reply stable/nats/templates/deployment.yaml]: ups...done

**Comment by carrodher** [comment]
I would add the comment related to the new version:
```
## Nats filenames:
##  - For Nats 1.x.x version, some filenames (binary, configuration file, pid file) uses `gnatsd` as part of the name.
##  - For Nats 2.x.x version, those filenames now uses `nats-server`
## In order to make the chart compatible with NATS versions 1.0.0 and 2.0.0 we have parametrized the following value
## to specify the proper filename according to the image version.
##
```
in the README, the rest LGTM

**Pull request merged**